float32[] armor_0
float32[] armor_1